### PR TITLE
feat: add dev suffix to version if not on a git tag

### LINKED ISSUE #7325: Differentiate between tagged releases and source builds in `--version` output
### Describe the feature

We can set an environment variable in the build script to append '-dev' at the end of the version string when the binary has not been built on a git tag.

### Additional context

- for example, rustc:
```text
$ rustc +stage2 -V
rustc 1.79.0-dev

$ rustc +stage2 -Vv
rustc 1.79.0-dev
binary: rustc
commit-hash: unknown
commit-date: unknown
host: x86_64-unknown-linux-gnu
release: 1.79.0-dev
LLVM version: 18.1.2
```

### PR DESCRIPTION
**Pull request by Colerar**
Fix #7325

Also checks if git working tree is dirty.

### CODE REVIEW

File: crates/node-core/build.rs [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
+
+    let sha = env::var("VERGEN_GIT_SHA")?;
+
+    let is_dirty = env::var("VERGEN_GIT_DIRTY")? == "true";
+    // > git describe --always --tags
+    // if not on a tag: v0.2.0-beta.3-82-g1939939b
+    // if on a tag: v0.2.0-beta.3
+    let not_on_tag = env::var("VERGEN_GIT_DESCRIBE")?.ends_with(&format!("-g{sha}"));
+    let is_dev = is_dirty || not_on_tag;
+    println!("cargo:rustc-env=RETH_DEV_STR={}", if is_dev { "-dev" } else { "" });
```
Comment (onbjerg) [inline crates/node-core/build.rs:25]: can we just call this `RETH_VERSION_SUFFIX`?

**Review by onbjerg — APPROVED** [review]
nit otherwise lgtm

File: crates/node-core/build.rs (hunk 1 above)
Comment (Colerar) [reply crates/node-core/build.rs]: renamed

**Review by onbjerg — APPROVED** [review]
lgtm

**Pull request merged**